### ISSUE TITLE
Use clinFHIR tool to display data in our server

### ISSUE THREAD

**Issue by kbertodatti** [chronic-care/mcc-project#27]

**Comment by kbertodatti** [comment]
Tool has limitations and cannot be used to create new resources.

**Comment by savanahmueller** [comment]
Implemented sort functionality for Goals; troubleshooting some errors encountered. -2/1

**Comment by kbertodatti** [comment]
Dan said that there was some code that can be used to implement these. Sravan is exploring.

**Comment by savanahmueller** [comment]
<img width="1320" src="https://api.zenhub.com/attachedFiles/eyJfcmFpbHMiOnsibWVzc2FnZSI6IkJBaHBBK3ZBQXc9PSIsImV4cCI6bnVsbCwicHVyIjoiYmxvYl9pZCJ9fQ==--a67fe0e8be9fd1a2020902987080a5e158c9425d/image.png" alt="image.png" />
Sravan is working to sort and filter the rows vs. the data points. Need to map the rows.

**Comment by mattStorer** [comment]
sort UI appears, but appears to have under "Filter Options" _fifteen_ identical instances of the "OHSU - POC" filter:

![image](https://github.com/chronic-care/mcc-project/assets/979938/942d95e7-0b90-4484-98a5-e56c1c1bca3a)

this is related to many warnings that appear in the console:

![image](https://github.com/chronic-care/mcc-project/assets/979938/b40061ca-bbcd-4b7d-a26f-7194abb1e626)

irrespective of whichever "OHSU - POC" filter I select, they all toggle the first instance - all others do nothing -

![image](https://github.com/chronic-care/mcc-project/assets/979938/1dc8032b-0f13-443b-ae5b-94df82b31018)

If I filter by "OHSU - POC" no records are displayed:

![image](https://github.com/chronic-care/mcc-project/assets/979938/a54a496a-5614-4f79-87ff-d50a0cb8e958)

There are "OHSU - POC" records available so the filter appears to not be working :

![image](https://github.com/chronic-care/mcc-project/assets/979938/4b70ea8e-96be-46e5-9a37-e0456cbebd13)

On a minor note, the "SORT/FILTER" UI always opens with no options selected, irrespective of whether or not I had previously selected an option, and which is presently driving record sorting or filtering.

**Comment by mattStorer** [comment]
yeah that's it.  the Medications page has 21 medications on there, and it has 21 instances of the "OHSU - POC" filter.  so evidently there's a loop in the code that iterates over resources, and is just adding the "source" to the filter list without checking for duplicates.

fixing this will address the UI issue.  still need to sort out why enabling a filter causes no records to show.

![image](https://github.com/chronic-care/mcc-project/assets/979938/8dbb95db-1fa5-456a-b8b2-f546785a5a76)

**Comment by mattStorer** [comment]
just checking status of the sort/filter logic in version 2.2.0.

firstly, the sort/filter UI looks WAY better, so thank you for those fixes!  the filtering also works as intended, so that's great.

that's the good news.

the bad news is that the sorting is still sorting only within the context of source groups, and not all together.  that is, all OHSU resources are sorted first (A-Z or Z-A), then (for example) Cerner resources are sorted (A-Z or Z-A).

screenshots:

A-Z, OHSU then Cerner:

![image](https://github.com/chronic-care/mcc-project/assets/979938/8a1b4620-d6d2-4f9c-9f72-617414c47234)

Z-A, OHSU then Cerner:
![image](https://github.com/chronic-care/mcc-project/assets/979938/0787220b-3d3c-4c52-a141-ea172b40a530)

so it looks like a little more needs to be done on this to get it where it needs to be.
